recipes/upstream/google-cloud/gke/main.tf

```diff
@@ -1,47 +1,80 @@
-module "google-kubernetes-engine" {
-  source               = "../../../../modules/distribution/gke"
-  prefix               = var.prefix
-  project_id           = var.project_id
-  region               = var.region
-  kube_config_filename = var.kube_config_filename
-  #  vpc        = var.vpc
-  #  subnet     = var.subnet
+locals {
+  kc_path = var.kube_config_path != null ? var.kube_config_path : path.cwd
+  kc_file = var.kube_config_filename != null ? "${local.kc_path}/${var.kube_config_filename}" : "${local.kc_path}/${var.prefix}_kube_config.yml"
+}
+
+module "google_kubernetes_engine" {
+  source     = "../../../../modules/distribution/gke"
+  prefix     = var.prefix
+  project_id = var.project_id
+  region     = var.region
+  #  vpc                = var.vpc
+  #  subnet             = var.subnet
   #  cluster_version    = var.cluster_version
-  #  instance_count = var.instance_count
+  #  instance_count     = var.instance_count
   #  instance_disk_size = var.instance_disk_size
   #  disk_type          = var.disk_type
   #  image_type         = var.image_type
   #  instance_type      = var.instance_type
 }
 
-resource "helm_release" "ingress-nginx" {
-  name             = "ingress-nginx"
-  repository       = "https://kubernetes.github.io/ingress-nginx"
-  chart            = "ingress-nginx"
-  create_namespace = true
-  namespace        = "ingress-nginx"
+resource "null_resource" "first_setup" {
+  depends_on = [module.google_kubernetes_engine.kubernetes_cluster_node_pool]
+
+  provisioner "local-exec" {
+    command = "sh ${path.cwd}/first_setup.sh"
+  }
+}
+
+resource "local_file" "kube_config_yaml" {
+  depends_on = [null_resource.first_setup]
+
+  content = templatefile("../../../../modules/distribution/gke/kubeconfig.yml.tmpl", {
+    cluster_name    = module.google_kubernetes_engine.cluster_name,
+    endpoint        = module.google_kubernetes_engine.cluster_endpoint,
+    cluster_ca      = module.google_kubernetes_engine.cluster_ca_certificate,
+    client_cert     = module.google_kubernetes_engine.client_certificate,
+    client_cert_key = module.google_kubernetes_engine.client_key
+  })
+  file_permission = "0600"
+  filename        = local.kc_file
+}
+
+provider "kubernetes" {
+  config_path = local_file.kube_config_yaml.filename
+}
+
+provider "helm" {
+  kubernetes {
+    config_path = local_file.kube_config_yaml.filename
+  }
+}
```

Comment:
Because the provider must be initialized after some resources are created and not in the first stages of the deployment of recipes.
Otherwise, it initializes an invalid value and fails to deploy the K8s resources (even if the kubeconfig file is generated).